airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/StateDecoratingIterator.java

```diff
@@ -27,10 +28,16 @@ public class StateDecoratingIterator extends AbstractIterator<AirbyteMessage> im
   private final JsonSchemaPrimitive cursorType;
   private final int stateEmissionFrequency;
 
+  private final String initialCursor;
   private String maxCursor;
-  private AirbyteMessage intermediateStateMessage;
   private boolean hasEmittedFinalState;
-  private int recordCount;
+
+  // The intermediateStateMessage is set to the latest state message.
+  // For every stateEmissionFrequency messages, emitIntermediateState is set to true and
+  // the latest intermediateStateMessage will be emitted.
+  private int totalRecordCount = 0;
+  private boolean emitIntermediateState = false;
+  private AirbyteMessage intermediateStateMessage = null;
```

Comment:
How come we're setting the `intermediateStateMessage = null` here over the previous code where it was originally `intermediateStateMessage`? Is it mainly to be explicit with the default value?

Reply:
Yes. I just want to make it explicit.